cmd/ipfs/daemon.go

```diff
@@ -181,6 +181,13 @@ func defaultMux(path string) corehttp.ServeOption {
 var fileDescriptorCheck = func() error { return nil }
 
 func daemonFunc(req cmds.Request, res cmds.Response) {
+	// Inject metrics before we do anything
+
+	err := mprome.Inject()
+	if err != nil {
+		log.Warningf("Injecting prometheus handler for metrics failed with message: %s\n", err.Error())
```

Comment:
Yes we can have flag, lacking metrics won't hurt working of the daemon so IMO it is better to continue.

If you want I can do Error log here instead.